lib/charms/opensearch/v0/opensearch_users.py

```diff
@@ -81,7 +81,7 @@ def create_role(
             f"{ROLE_ENDPOINT}/{role_name}",
             payload={**(permissions or {}), **(action_groups or {})},
         )
-        if resp.get("status") != "OK":
+        if resp.get("status") not in ["CREATED", "OK"]:
```

Comment:
Why 2 potential statuses? one for role create and the other update? if that's the case, I don't think we should handle the update, right?

Reply:
It was originally intended to handle updates when we call it twice due to an event deferral, but there's no `defer` calls after this function in `_on_index_created`, so it shouldn't be necessary.